### PR TITLE
feat: Add a `collapse` method to `opensearchpy.helpers.search.Search`

### LINKED ISSUE #408: [FEATURE] Add a `collapse` method to `opensearchpy.helpers.search.Search`
### Is your feature request related to a problem?
The only way to currently add a *collapse* requirement to a search body is to use the `extra()` method. This is not the cleanest way because as mentioned in its docstring, `extra()` is mostly here for backward compatibility.

### What solution would you like?
A solution would be to treat it the same way as `sort`:
* Add a `collapse()` method `opensearchpy.helpers.search.Search` which creates and stores the `dict` in a `self._collapse` attribute.
* Add `self._collapse`, if defined, in the body of the request in the `to_dict()` method.

### PR DESCRIPTION
**Pull request by qcoumes**
### Description
This PR adds a `collapse` method to `opensearchpy.helpers.search.Search` which allows adding a `collapse` requirement to a query without the need of `.extra()`.

This is achieved similarly to how `sort()` is done.

### Issues Resolved
#408

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Added 'point in time' APIs to the pyi files in sync and async client ([#378](https://github.com/opensearch-project/opensearch-py/pull/378))
 - Added MacOS and Windows CI workflows ([#390](https://github.com/opensearch-project/opensearch-py/pull/390))
 - Compatibility with OpenSearch 2.1.0 - 2.6.0 ([#381](https://github.com/opensearch-project/opensearch-py/pull/381))
+- Added a `collapse` method to `opensearchpy.helpers.search.Search` ([#408](https://github.com/opensearch-project/opensearch-py/issues/408))
```
Comment (saimedhi) [inline CHANGELOG.md:12]: Hello @qcoumes, just a small correction. Please change PR No. to 409

Comment (qcoumes) [reply CHANGELOG.md]: Done !

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Added 'point in time' APIs to the pyi files in sync and async client ([#378](https://github.com/opensearch-project/opensearch-py/pull/378))
 - Added MacOS and Windows CI workflows ([#390](https://github.com/opensearch-project/opensearch-py/pull/390))
 - Compatibility with OpenSearch 2.1.0 - 2.6.0 ([#381](https://github.com/opensearch-project/opensearch-py/pull/381))
+- Added a `collapse` method to `opensearchpy.helpers.search.Search` ([#409](https://github.com/opensearch-project/opensearch-py/issues/409))
```
Comment (dblock) [inline CHANGELOG.md:12]: Nit: "Added Search#collapse" is probably more straightforward.

**Review by dblock — COMMENTED** [review]
Looks good! Add something to the users guide and resolve CHANGELOG conflict?

**Comment by qcoumes** [comment]
I resolved the conflict.

I don't know where I should add some documentation in the user guide. The *Using DSL features from opensearch-dsl-py* chapter is mainly a redirection to `opensearch-dsl-py`'s documentation.

It seems weird to have a subchapter about `collapse` without some documentation about basic features.

**Comment by dblock** [comment]
> It seems weird to have a subchapter about `collapse` without some documentation about basic features.

Yes, you're of course right. Would love someone (hint hint) to contribute some basic features doc. I'll let it slide for this PR :)

Thank you!

**Review by dblock — DISMISSED** [review]

**Review by dblock — APPROVED** [review]

**Pull request merged**